rskj-core/src/main/java/org/ethereum/listener/GasPriceTracker.java

```diff
@@ -90,7 +97,9 @@ private void onTransaction(Transaction tx) {
 
     public synchronized Coin getGasPrice() {
         if (window[0] == null) { // not filled yet
-            return defaultPrice;
+            return Optional.ofNullable(blockchain.getBestBlock())
```

Comment:
yeah, this is one way to do it (simplest one). But we could also load more blocks, e.g. in the ctor. (parent of the best one, parent of parent and so on..) from the db until the `window` array is filled up (the algorithm uses this array of 512 items as a main source of historical data [here](https://github.com/rsksmart/rskj/blob/91e5f60de29160ca8b96542bd0e59b0a57c31f13/rskj-core/src/main/java/org/ethereum/listener/GasPriceTracker.java#L96-L98)). And [this](https://github.com/rsksmart/rskj/blob/91e5f60de29160ca8b96542bd0e59b0a57c31f13/rskj-core/src/main/java/org/ethereum/listener/GasPriceTracker.java#L65-L76) is a place where that array is being populated.

Reply:
Addressed 👍